### PR TITLE
[TH2-2273] Do not publish empty messages batches

### PR DESCRIPTION
**Pull request by OptimumOpium**

### CODE REVIEW

File: src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.function.Executable
+import org.mockito.kotlin.any
+import org.mockito.kotlin.anyOrNull
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
```
Comment (Nikita-Smirnov-Exactpro) [inline src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt:44]:
```suggestion
import org.mockito.kotlin.times
import org.mockito.kotlin.verify
```

File: src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+        @Test
+        fun `publishes to the correct pin according to attributes`() {
+            val batch = MessageGroupBatch.newBuilder()
+                .addGroups(MessageGroup.newBuilder()
+                    .apply { this += message("test-message", Direction.FIRST, "test-alias") }
+                ).build()
+            router.send(batch, "test")
+
+            val captor = argumentCaptor<ByteArray>()
+            verify(connectionManager).basicPublish(eq("test-exchange"), eq("test2"), anyOrNull(), captor.capture())
```
Comment (Nikita-Smirnov-Exactpro) [inline src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt:120]:
```suggestion
            verify(connectionManager, times(1)).basicPublish(eq("test-exchange"), eq("test2"), anyOrNull(), captor.capture())
```

File: src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        fun `publishes to all correct pin according to attributes`() {
+            val batch = MessageGroupBatch.newBuilder()
+                .addGroups(MessageGroup.newBuilder()
+                    .apply { this += message("test-message", Direction.FIRST, "test-alias") }
+                ).build()
+            router.sendAll(batch)
+
+            val captor = argumentCaptor<ByteArray>()
+            verify(connectionManager).basicPublish(eq("test-exchange"), eq("test"), anyOrNull(), captor.capture())
+            verify(connectionManager).basicPublish(eq("test-exchange"), eq("test2"), anyOrNull(), captor.capture())
```
Comment (Nikita-Smirnov-Exactpro) [inline src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt:169]:
```suggestion
            verify(connectionManager, times(1)).basicPublish(eq("test-exchange"), eq("test"), anyOrNull(), captor.capture())
            verify(connectionManager, times(1)).basicPublish(eq("test-exchange"), eq("test2"), anyOrNull(), captor.capture())
```

File: src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+                )
+            )
+        )
+
+        @Test
+        fun `subscribes to correct queue`() {
+            val monitor = router.subscribe(mock { }, "1")
+            Assertions.assertNotNull(monitor) { "monitor must not be null" }
+
+            verify(connectionManager).basicConsume(eq("queue1"), any(), any())
```
Comment (Nikita-Smirnov-Exactpro) [inline src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt:218]:
```suggestion
            verify(connectionManager, times(1)).basicConsume(eq("queue1"), any(), any())
```

File: src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        }
+
+        @Test
+        fun `subscribes to all matched queues`() {
+            val monitor = router.subscribeAll(mock { })
+            Assertions.assertNotNull(monitor) { "monitor must not be null" }
+
+            verify(connectionManager).basicConsume(eq("queue1"), any(), any())
+            verify(connectionManager).basicConsume(eq("queue2"), any(), any())
```
Comment (Nikita-Smirnov-Exactpro) [inline src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt:227]:
```suggestion
            verify(connectionManager, times(1)).basicConsume(eq("queue1"), any(), any())
            verify(connectionManager, times(1)).basicConsume(eq("queue2"), any(), any())
```

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

File: src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt (hunk 2 above)
Comment (OptimumOpium) [reply src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt:120]: It adds the 1 implicitly. No needs to do it manually

**Pull request merged**